File: 27.c
//Project Euler solution 27, author: http://github.com/TerkaSlan
#include <stdio.h>
#include <stdbool.h>
#include <time.h>
#include <inttypes.h>

bool isPrime(int32_t expression)
{
  if (expression <= 0)
    return false;

  for(uint32_t i = 2; i*i <= expression; i++){
    if(expression % i == 0){
      return false;
    }
  }
  return true;
}

int main(int argc, char const *argv[]) {
  clock_t start = clock();
  int16_t const lower_limit = -1000;
  int16_t const upper_limit = 1000;

  uint32_t max_count = 0;
  int32_t product_of_the_coef = 0;
  int32_t n = 0;

  for (int32_t a = lower_limit + 1; a < upper_limit; a++) {
    for (int32_t b = lower_limit + 1; b < upper_limit; b++) {
      for(n = 0; isPrime(n*n + a*n + b); n++);
      if(n > max_count){
        max_count = n;
        product_of_the_coef = a*b;
      }
    }
  }
  clock_t end = clock();
  printf("Product: %" PRId32, product_of_the_coef);
  printf("\nTook me %f secs\n", ((end-start)/CLOCKS_PER_SEC)*1.0);
  return 0;
}

File: 27.py
#Project Euler solution 27, author: http://github.com/TerkaSlan #

from time import clock

def isPrime(expr):
    if expr <= 0:
        return False
    # trying to speed up for most of the non-prime numbers,
    # suprisingly, got 1s out of it
    if expr % 2 == 0 or expr % 3 == 0 or expr % 5 == 0:
        return False

    i = 7;
    while i*i <= expr:
        if expr % i == 0:
            return False
        i += 1

    return True

start = clock()

max_streak = 0
max_product = 0
lower_limit = -999
upper_limit = 1000

for a in range(lower_limit,upper_limit,1):
    # b cannot be negative, it will be immediatelly rejected by isPrime when n = 0
    for b in range(0,upper_limit,1):
        n = 0
        while(isPrime(n*n + a*n + b)):
            n += 1
        #print 'For a=', a, ' b=', b, ' seq found: ', n
        if n > max_streak:
            max_streak = n
            max_product = a*b

end = clock()
print 'Found: ', max_product
print 'Took me ', end-start, 'secs' # 3s
